features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/resources/ResourceResolver.kt

```diff
@@ -0,0 +1,284 @@
+    ) {
+        bitmapCachesManager.registerCallbacks(applicationContext)
+
+        val resourceId = tryToGetResourceFromCache(drawable = drawable)
+
+        if (resourceId != null) {
+            // if we got here it means we saw the bitmap before,
+            // so we don't need to send the resource again
+            resourcesSerializerCallback.onSuccess(resourceId)
+            return
+        }
+
+        val bitmapFromDrawable = if (drawable is BitmapDrawable && shouldUseDrawableBitmap(drawable)) {
+            drawable.bitmap // cannot be null - we already checked in shouldUseDrawableBitmap
+        } else {
+            null
+        }
+
+        // do in the background
+        threadPoolExecutor.executeSafe("resolveResourceId", logger) {
+            @Suppress("ThreadSafety") // this runs inside an executor
+            createBitmapAsync(
+                resources = resources,
+                drawable = drawable,
+                drawableWidth = drawableWidth,
+                drawableHeight = drawableHeight,
+                displayMetrics = displayMetrics,
+                bitmapFromDrawable = bitmapFromDrawable,
+                resolveResourceCallback = object : ResolveResourceCallback {
+                    override fun onResolved(resourceId: String, byteArray: ByteArray) {
+                        resourceItemCreationHandler.queueItem(resourceId, byteArray)
+                        resourcesSerializerCallback.onSuccess(resourceId)
+                    }
+
+                    override fun onFailed() {
+                        resourcesSerializerCallback.onFailure()
+                    }
+                }
+            )
+        }
+    }
+
+    // endregion
+
+    // region private
+
+    @WorkerThread
+    private fun createBitmapAsync(
+        resources: Resources,
+        drawable: Drawable,
+        drawableWidth: Int,
+        drawableHeight: Int,
+        displayMetrics: DisplayMetrics,
+        bitmapFromDrawable: Bitmap?,
+        resolveResourceCallback: ResolveResourceCallback
+    ) {
+        var handledBitmap: Bitmap? = null
+        if (bitmapFromDrawable != null) {
+            handledBitmap = tryToGetBitmapFromBitmapDrawable(
+                drawable = drawable as BitmapDrawable,
+                bitmapFromDrawable = bitmapFromDrawable,
+                resolveResourceCallback = resolveResourceCallback
+            )
+        }
+
+        if (handledBitmap == null) {
+            tryToDrawNewBitmap(
+                resources = resources,
+                drawable = drawable,
+                drawableWidth = drawableWidth,
+                drawableHeight = drawableHeight,
+                displayMetrics = displayMetrics,
+                resolveResourceCallback = resolveResourceCallback
+            )
+        }
+    }
+
+    @Suppress("ReturnCount")
+    @WorkerThread
+    private fun resolveResourceHash(
+        drawable: Drawable,
+        bitmap: Bitmap,
+        byteArray: ByteArray,
```

Comment:
we should give a more clear name to this one, currently by looking on the name it is not clear what this parameter represents

Reply:
agreed - I've renamed it